src/tests/test_config.cpp

```diff
@@ -286,4 +289,51 @@ BOOST_AUTO_TEST_CASE ( test_variable_info )
 	}
 }
 
+BOOST_AUTO_TEST_CASE (
+		add_child_ThisIsEmpty_newKey_AppendEmptyChildToThis_ReturnNewChild )
+{
+	config actual;
+	const config new_child = actual.add_child("A");
+	const config expected("A");
+	// Assert add_child adds the child correctly
+	BOOST_CHECK_EQUAL (actual, expected);
+	// Assert the new child returned by config::add_child is correct
+	BOOST_CHECK_EQUAL (new_child, config());
```

Comment:
Your indentation and spacing is better than the result of running clang-format on this file, however running clang-format and then cherry-picking the good differences would be better than both. For example, our style is not to have a space before the bracket in `if(` or `config(`, so it makes sense to omit it from `BOOST_CHECK_EQUAL(` too.

Reply:
I make my IDE use clang-format to format the code.